hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -260,9 +260,12 @@ public static void processFiles(FileSystem fs, String basePathStr, Function<File
     if (assumeDatePartitioning) {
       return getAllPartitionFoldersThreeLevelsDown(fs, basePathStr);
```

Comment:
Why does this not list from metadata table?

Reply:
Good point. This also needs to move inside. Will make the adjustment.

Reply:
So this already happens inside `FileSystemBackedTableMetadata`. Removing the if block.